### DISCUSSION TITLE
Split building into multiple jobs and them release them as one

### DISCUSSION THREAD

**Discussion by grongor** [goreleaser/goreleaser#2472] [Q&A] [answered]
Hello, I'm wondering if it's possible to split the "building" part of the release into multiple jobs in GitHub actions, and then when all builds are done, stitch them together into one single release containing all the builds.

What I want to achieve is to build the same binary (and/or archives) against different versions of a system library. To be precise, I need the library `libsnmp` for my project and on different OSes it has different versions, so to simplify my life (and possibly the lives of other users), I would like to build the binary against `libsnmp30` (old LTS Debian), `libsnmp35` (current LTS Ubuntu) and `libsnmp40` (current LTS Debian). So I would need 3 separate jobs, each with a different version of the library installed, and then run the `goreleaser build` there for each one, and later in the final job just `goreleaser release` them together.

Is this possible somehow?

edit: just found a similar question, I bet the solution will be basically the same https://github.com/goreleaser/goreleaser/discussions/2440

**Comment by caarlos0** [comment, marked as answer]
sorry, there's no way of doing that right now...

**Comment by grongor** [comment]
Oh, that is unfortunate. Thank you for the fast reply anyway! I hope this feature will at least make it to the roadmap so I can improve the builds someday :)

**Reply by caarlos0** [reply]
fwiw, this has been just released as a pro feature https://carlosbecker.com/posts/goreleaser-prebuilt/

hit me up if you want to try, I'll be happy to give you a discount :)

**Reply by grongor** [reply]
Hey, thanks for letting me know. It's not perfect, still too many things to do "manually", but good that it's at least possible. I'd love to give it a try/use it, but unfortunately, I'm not willing to pay for the tool (unless the discount would be significant) because I'd just use it for a few of my open-source packages which generate exactly zero profits for me (I do it for the community). Don't take it wrong, I personally think that the "Pro" model is amazing and I encourage it...but for enterprise/private use. I just can't pay for tools to provide something for free to everybody.
